lighthouse-core/audits/seo/font-size.js

```diff
@@ -0,0 +1,233 @@
+  if (styleDeclaration && styleDeclaration.type === CSSStyleDeclaration.Type.Regular) {
+    const startLine = styleDeclaration.range ? styleDeclaration.range.startLine : 0;
+    const startColumn = styleDeclaration.range ? styleDeclaration.range.startColumn : 0;
+    return `${styleDeclaration.styleSheetId}@${startLine}:${startColumn}`;
+  } else {
+    return `node_${node.nodeId}`;
+  }
+}
+
+class FontSize extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'font-size',
+      description: 'Document uses legible font sizes.',
+      failureDescription: 'Document doesn\'t use legible font sizes.',
+      helpText: 'Font sizes less than 16px are too small to be legible and require mobile ' +
+      'visitors to “pinch to zoom” in order to read. ' +
+      '[Learn more](https://developers.google.com/speed/docs/insights/UseLegibleFontSizes).',
+      requiredArtifacts: ['FontSize', 'URL', 'Viewport'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const hasViewportSet = ViewportAudit.audit(artifacts).rawValue;
+    if (!hasViewportSet) {
+      return {
+        rawValue: false,
+        debugString: 'Text is illegible because of a missing viewport config',
+      };
+    }
+
+    const {
+      analyzedFailingNodesData,
+      analyzedFailingTextLength,
+      failingTextLength,
+      visitedTextLength,
+      totalTextLength,
+    } = artifacts.FontSize;
+
+    if (totalTextLength === 0) {
+      return {
+        rawValue: true,
+      };
+    }
+
+    const failingRules = getUniqueFailingRules(analyzedFailingNodesData);
+    const percentageOfPassingText =
+      (visitedTextLength - failingTextLength) / visitedTextLength * 100;
+    const pageUrl = artifacts.URL.finalUrl;
+
+    const headings = [
+      {key: 'source', itemType: 'url', text: 'Source'},
+      {key: 'selector', itemType: 'code', text: 'Selector'},
+      {key: 'coverage', itemType: 'text', text: 'Coverage'},
+      {key: 'fontSize', itemType: 'text', text: 'Font Size'},
+    ];
+
+    const tableData = failingRules.sort((a, b) => b.textLength - a.textLength)
+      .map(({cssRule, textLength, fontSize, node}) => {
+        const percentageOfAffectedText = textLength / visitedTextLength * 100;
+        const origin = findStyleRuleSource(pageUrl, cssRule, node);
+
+        return {
+          source: origin.source,
+          selector: origin.selector,
+          coverage: `${percentageOfAffectedText.toFixed(2)}%`,
+          fontSize: `${fontSize}px`,
+        };
+      });
+
+    // all failing nodes which were not fully analyzed will be shown in the 'Other' category
+    if (analyzedFailingTextLength < failingTextLength) {
+      const percentageOfUnanalyzedFailingText =
+        (failingTextLength - analyzedFailingTextLength) / visitedTextLength * 100;
+
+      tableData.push({
+        source: 'Other',
```

Comment:
I've been thinking perhaps "Other" is a bit misleading here. Because we didn't analyze the nodes, it very well could be that the source is the same as one in the table, but we just don't know it. Is that right? 

If so, maybe we say something like "Unknown" instead?

Reply:
"Unknown" sounds like something we were not able to recognize. How about "Not analyzed"?

Reply:
yeah I see what ya mean, @paulirish had some suggestions here

* Other -> Additional unanalyzed small text
* Add a row to the table for "Legible text" to make the table add up to 100%
* Change "Coverage" to "% of Page Text"

Reply:
Updated!

Top:
<img width="725" alt="screen shot 2017-12-06 at 17 02 31" src="https://user-images.githubusercontent.com/985504/33671265-6ef4539c-daa7-11e7-9a7e-f996b40eb2ac.png">

Bottom:
<img width="734" alt="very-bottom-2" src="https://user-images.githubusercontent.com/985504/33671581-458e9444-daa8-11e7-9a48-a010a2f8dd53.png">


Some thoughts/questions:
- "Additional unanalyzed small text" is waaay too long, IMO we should replace it with something shorter.
- I added "≥ 16px" for consistency, but not sure if we should keep it. WDTY?

Reply:
> Addtl illegible text

Reply:
> I added "≥ 16px" for consistency, but not sure if we should keep it. WDTY?
 
love it. nice work.